src/NUnitFramework/nunitlite/TextRunner.cs

```diff
@@ -318,21 +318,21 @@ public void ReportResults(ITestResult result)
             _textUI.DisplaySummaryReport(Summary);
         }
 
-        private int ExploreTests()
+        private int ExploreTests(ITestFilter filter)
         {
 #if !PORTABLE
             ITest testNode = _runner.LoadedTest;
```

Comment:
I think this line is the key to the problem I pointed out. In our online discussion, we talked about having NUnitLite actually call an Explore method on the runner, with a filter applied. If testNode were already filtered, then none of the other stuff would be needed. Does this seem feasible to you?